modules/apm/src/main/java/org/elasticsearch/telemetry/apm/internal/metrics/APMMetric.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.telemetry.apm.internal.metrics;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.metrics.Meter;
+
+import org.elasticsearch.common.component.AbstractLifecycleComponent;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.telemetry.metric.DoubleCounter;
+import org.elasticsearch.telemetry.metric.DoubleGauge;
+import org.elasticsearch.telemetry.metric.DoubleHistogram;
+import org.elasticsearch.telemetry.metric.DoubleUpDownCounter;
+import org.elasticsearch.telemetry.metric.LongCounter;
+import org.elasticsearch.telemetry.metric.LongGauge;
+import org.elasticsearch.telemetry.metric.LongHistogram;
+import org.elasticsearch.telemetry.metric.LongUpDownCounter;
+import org.elasticsearch.telemetry.metric.MetricName;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.elasticsearch.telemetry.apm.internal.APMAgentSettings.APM_ENABLED_SETTING;
+
+public class APMMetric extends AbstractLifecycleComponent implements org.elasticsearch.telemetry.metric.Metric {
+    private static final Meter NOOP = OpenTelemetry.noop().getMeter("noop");
+    private final Instruments instruments = new Instruments(NOOP);
+    private volatile boolean enabled;
+    private final AtomicReference<APMServices> services = new AtomicReference<>();
+
+    record APMServices(Meter meter, OpenTelemetry openTelemetry) {}
+
+    public APMMetric(Settings settings) {
+        this.enabled = APM_ENABLED_SETTING.get(settings);
+    }
+
+    public void setEnabled(boolean enabled) {
```

Comment:
it is here wired to dynamic cluster settings
 https://github.com/elastic/elasticsearch/pull/99832/files#diff-8e5ca1315006cc00c58ef42d4bfc14f50b30d11567a845f8e551ecf3e52c830bR58
regarding concurrency considerations we concluded that:
1. enabled should be volatile
2. Instruments.setProvider method is acquiring a lock. It needs this so that new instruments are not added at the same time when the implementation of the `Meter` (noop/otel) is set
3. whenever we add a new instrument via register* methods we need to acquire a lock
adding new instruments calling setProvider should be rare (we expect those to be called at most once during node lifetime)